src/solver/infeasible-problem-analysis/problem.cpp

```diff
@@ -11,80 +11,179 @@ namespace Antares
+
+void VariablesBoundsAnalysis::run()
+{
+    for (auto& var : problem_->variables())
     {
-        throw SlackVariablesEmpty(
-          "Cannot generate infeasibility report: no constraints have been selected");
+        double lowBound = var->lb();
+        double upBound = var->ub();
+        std::string name = var->name();
+        if (lowBound > upBound)
+        {
+            storeIncorrectVariable(name, lowBound, upBound);
+        }
     }
-    buildObjective();
-    const MPSolver::ResultStatus status = Solve();
-    if ((status != MPSolver::OPTIMAL) && (status != MPSolver::FEASIBLE))
+
+    if (foundIncorrectVariables())
+        hasDetectedInfeasibilityCause_ = true;
+}
+
+void VariablesBoundsAnalysis::storeIncorrectVariable(std::string name, double lowBound, double upBound)
+{
+    incorrectVars_.push_back(VariableBounds(name, lowBound, upBound));
+}
+
+bool VariablesBoundsAnalysis::foundIncorrectVariables()
+{
+    return (incorrectVars_.size() != 0);
+}
+
+void VariablesBoundsAnalysis::printReport()
+{
+    for (auto& var : incorrectVars_)
+    {
+        logs.notice() << var.name << " : low bound = " << var.lowBound << ", up bound = " << var.upBound;
+    }
+}
+
+
+// ===============================
+// Unfeasibility analyzer 
+// ===============================
+
+// gp : this class should be renamed into UnfeasibilityAnalyzer
+
+InfeasibleProblemAnalysis::InfeasibleProblemAnalysis(const std::string& solverName, const PROBLEME_SIMPLEXE_NOMME* ProbSpx)
+{
+    // gp : Here we have a dependency on PROBLEME_SIMPLEXE_NOMME and MPSolver.
+    // gp : We should Convert() the PROBLEME_SIMPLEXE_NOMME into a MPSolver outside (and before) this constructor.
+    // gp : And so we should have a MPSolver* passed here.
+    // gp : It would be easier to test this class in isolation.
+    problem_ = std::shared_ptr<MPSolver>(ProblemSimplexeNommeConverter(solverName, ProbSpx).Convert());
+
+    analysisList_.push_back(std::make_unique<VariablesBoundsAnalysis>(problem_));
+    analysisList_.push_back(std::make_unique<SlackVariablesAnalysis>(problem_));
+}
+
+void InfeasibleProblemAnalysis::run()
+{
+    logs.info();
+    logs.info() << "Solver: Starting unfeasibility analysis...";
+
+    for (auto& analysis : analysisList_)
     {
-        throw ProblemResolutionFailed(
-          "Linear problem could not be solved, and infeasibility analysis could not help");
+        logs.info();
+        logs.info() << analysis->title() << " : running...";
+        analysis->run();
+        if (!analysis->hasDetectedInfeasibilityCause())
+            logs.notice() << analysis->title() << " : nothing detected.";
+    }
+}
+
+void InfeasibleProblemAnalysis::printReport()
+{
+    for (auto& analysis : analysisList_)
+    {
+        if (analysis->hasDetectedInfeasibilityCause())
+        {
+            logs.info() << analysis->title() << " : printing report";
+            analysis->printReport();
+            return;
+        }
```

Comment:
As discussed, the code looks much clearer without a list
```suggestion
    VariableBoundsChecker varChecker(problem_);
    if (varChecker.infeasibilityDetected())
    {
         // Handle variable infeasibility, for example
         varChecker.printReport();
    }
    else
    {
        ConstraintCheckerUsingSlackVariables constraintChecker(problem_);
        if (constraintChecker.infeasibilityDected())
        {
            // Handle this infeasibility, for example
            constraintChecker.printReport();
        }
   }
```

Reply:
I'm sorry to disagree, for the following reasons : 
- In this code suggested here, this is a duplication : both analysis are called in the same way.
- The analyzer depends on the two analysis, and cannot properly tested in isolation.
- The analyzer'interface has changed, the analysis part is no more separated from reporting part. I guess it's a question of taste : client of the analyzer could prefer (or not) that the 2 steps are done in one glance.